probe-rs/src/architecture/arm/sequences.rs

```diff
@@ -552,11 +549,6 @@ pub trait ArmDebugSequence: Send + Sync + Debug {
                         // At least 2 idle cycles (SWDIO/TMS Low).
                         interface.swj_sequence(3, 0x00)?;
                     }
-
-                    // SWD should now be activated, so we can try and connect to the debug port
-                    if self.debug_port_connect(interface, dp).is_ok() {
-                        return Ok(());
```

Comment:
Doesn’t this change now mean that we’ll always loop N times? (I don’t see where we break now)

Reply:
unless I got my scoping wrong, this was moved to after the match, just before continuing to the next loop iteration